src/plugins/intel_gpu/src/graph/loop.cpp

```diff
@@ -316,181 +381,275 @@ void loop_inst::update_mapped_memory() {
     }
 }
 
+
+void loop_inst::update_mapped_memory() {
+    if (!preproc_memories_done) {
+        return;
+    }
+
+    update_output_mapped_memory();
+    update_input_mapped_memory();
+    update_backedge_mapped_memory();
+}
+
 event::ptr loop_inst::set_output_memory(memory::ptr mem, bool check, size_t idx) {
     auto ev = primitive_inst::set_output_memory(mem, check, idx);
+    if (mem == nullptr) {
+        std::cout << "loop_inst::set_output_memory : " << mem << std::endl;
+        std::cout << "loop_inst::set_output_memory : " << mem->get_layout().to_short_string() << std::endl;
+    }
     update_mapped_memory();
     return ev;
 }
 
-void loop_inst::preprocess_output_memory() {
-    auto& engine = _network.get_engine();
+void loop_inst::preprocess_output_memory(const int64_t num_iterations) {
     concatenated_output_mem_mappings.reserve(_output_primitive_maps.size());
     for (size_t i = 0; i < _output_primitive_maps.size(); ++i) {
         const auto& output_mapping = _output_primitive_maps.at(i);
-        const primitive_id& external_id = output_mapping.external_id;
-        const primitive_id& internal_id = output_mapping.internal_id;
-        if (output_mapping.axis < 0) {
-            memory::ptr memory = get_external_memory(external_id);
-            body_network->get_primitive(internal_id)->set_output_memory(memory);
+        const auto& external_id = output_mapping.external_id;
+        const auto& internal_id = output_mapping.internal_id;
+        GPU_DEBUG_LOG << i << ") output mapping - external " << external_id.to_string() << std::endl;
```

Comment:
Please add some more info like "Preprocessing output memory for..."

Reply:
Is it necessary to add "preprocessing output memory" ? I don't add this comments because i think it is tautology for that GPU_DEBUG_LOG show the function name "preprocess_output_memory".

Reply:
Hm that make sense, got it.